### DISCUSSION TITLE
How to debug (or trace) EventLoop callbacks?

### DISCUSSION THREAD

**Discussion by vovkapoc** [revoltphp/event-loop#93] [Q&A] [answered]
Hello everybody! Sorry for the stupid question, but I'm new to EventLoop/Fiber/async in php. Please tell me how to debug the main EventLoop? For example, I get callback ids that are currently active using EventLoop::getIdentifiers(), using var_dump(EventLoop::getDriver()) I see the types of all fiber IDs. But how to understand by fiber ID which callback is tied to it?
It's interesting because I use some libraries from amphp and as a result I see 3 fiber's that I didn't create and which for some reason do not close on their own at the end of the program's cycle. I want to understand which callbacks are connected to these ids.

**_Sorry for my bad English, I from Ukraine!_**

**P. S. EventLoop use libuv as primary driver**

**Comment by kelunik** [comment, marked as answer]
Callback IDs aren't tied to fibers. You can use [`REVOLT_DRIVER_DEBUG_TRACE`](https://github.com/revoltphp/event-loop/blob/358572ca31647df7dbe930f13c8567136beffa33/src/EventLoop/DriverFactory.php#L46C22-L46C47) to use the `TracingDriver` to record additional debugging information. `dump` can be used to get info about the creation traces of still enabled callbacks: https://github.com/revoltphp/event-loop/blob/358572ca31647df7dbe930f13c8567136beffa33/src/EventLoop/Driver/TracingDriver.php#L192

**Reply by vovkapoc** [reply]
Thank you very much!😎👍
